packages/devtools_app/lib/src/inspector/flutter/inspector_screen_details_tab.dart

```diff
@@ -23,64 +24,96 @@ class InspectorDetailsTabController extends StatelessWidget {
+  TabController _tabControllerWithoutLayoutExplorer;
+
+  @override
+  void initState() {
+    super.initState();
+    addAutoDisposeListener(
+      _tabControllerWithLayoutExplorer =
+          TabController(length: _tabsLengthWithLayoutExplorer, vsync: this),
+    );
+    addAutoDisposeListener(
+      _tabControllerWithoutLayoutExplorer =
+          TabController(length: _tabsLengthWithoutLayoutExplorer, vsync: this),
     );
   }
 
   @override
   Widget build(BuildContext context) {
     final tabs = <Tab>[
       _buildTab('Details Tree'),
-      if (layoutExplorerSupported) _buildTab('Layout Explorer'),
+      if (widget.layoutExplorerSupported) _buildTab('Layout Explorer'),
     ];
     final tabViews = <Widget>[
-      detailsTree,
-      if (layoutExplorerSupported) LayoutExplorerTab(controller: controller),
+      widget.detailsTree,
+      if (widget.layoutExplorerSupported)
+        LayoutExplorerTab(controller: widget.controller),
     ];
+    final _tabController = widget.layoutExplorerSupported
+        ? _tabControllerWithLayoutExplorer
+        : _tabControllerWithoutLayoutExplorer;
     final focusColor = Theme.of(context).focusColor;
     return Container(
       decoration: BoxDecoration(
         border: Border.all(color: focusColor),
       ),
-      child: DefaultTabController(
-        length: tabs.length,
-        child: Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: <Widget>[
-            Container(
-              padding: const EdgeInsets.only(bottom: 8.0),
-              child: Row(
-                children: <Widget>[
-                  Flexible(
-                    child: Container(
-                      color: focusColor,
-                      child: TabBar(
-                        labelColor: Theme.of(context).textTheme.body1.color,
-                        tabs: tabs,
-                        isScrollable: true,
-                      ),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Container(
+            padding: const EdgeInsets.only(bottom: 8.0),
+            child: Row(
+              children: <Widget>[
+                Flexible(
+                  child: Container(
+                    color: focusColor,
+                    child: TabBar(
+                      controller: _tabController,
+                      labelColor: Theme.of(context).textTheme.body1.color,
+                      tabs: tabs,
+                      isScrollable: true,
                     ),
                   ),
-                  if (actionButtons != null)
-                    Expanded(
-                      child: actionButtons,
-                    ),
-                ],
-                mainAxisAlignment: MainAxisAlignment.spaceBetween,
-              ),
+                ),
+                if (widget.actionButtons != null &&
+                    _tabController.index != _layoutExplorerTabIndex)
```

Comment:
nit: instead of hiding for the layout explorer, only show for the details tree.
I think this is a details tree only feature.

Reply:
done